brownie/_expansion.py

```diff
@@ -0,0 +1,57 @@
+import re
+
+from dotenv.variables import parse_variables
+
+
+def expand_posix_vars(dct, variables):
+    """expand_posix_vars performs recursive POSIX-style variable expansion on a dictionary
+
+    This supports nested dictionaries.
+    """
+    if isinstance(dct, (dict,)):
+        for key, val in dct.items():
+            dct[key] = expand_posix_vars(val, variables)
+    elif isinstance(dct, (list,)):
+        for index in range(len(dct)):
+            dct[index] = expand_posix_vars(dct[index], variables)
+    elif isinstance(dct, (str,)):
+        dct = _str_to_python_value(_expand(dct, variables))
+    return dct
+
+
+def _expand(value, variables={}):
+    """_expand does POSIX-style variable expansion
+
+    This is adapted from python-dotenv, specifically here:
+
+    https://github.com/theskumar/python-dotenv/commit/17dba65244c1d4d10f591fe37c924bd2c6fd1cfc
+
+    We need this layer here so we can explicitly pass in variables;
+    python-dotenv assumes you want to use os.environ.
+    """
+
+    if value is None:
+        return None
+    if not isinstance(value, (str)):
+        return value
```

Comment:
The first check here seems redundant, I believe the 2nd, broader check would always catch the first condition.

Also the brackets around `str` have no effect without a trailing comma.